include/boost/geometry/util/for_each_dimension.hpp

```diff
@@ -0,0 +1,194 @@
+// Boost.Geometry
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_UTIL_FOR_EACH_DIMENSION_HPP
+#define BOOST_GEOMETRY_UTIL_FOR_EACH_DIMENSION_HPP
+
+
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/util/type_traits_std.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail
+{
+
+// NOTE: Out of curiosity I tested the compilation times of various implementations.
+//   Compilation times were tested for point and dimension 1000 with g++-6.3.0 and msvc-14.1.
+//   Each coordinate of a point was set to the dimension value.
+// In general we deal with small dimensions so the simplest O(N) should be the fastest
+//   one anyway.
+// It seems that simple O(N) version is faster to compile than O(logN) version based on
+//   intervals even for a big value such as 1000. Which is wierd and may indicate that I
+//   missed something.
+//   The simple version generates deeper recursion but this is not a problem since in general we
+//   deal with small number of dimensions.
+// A third version I tested using C++17 fold expression was faster in some cases.
+//   It depended on optimization level.
+// This utility was also slightly faster than for_each_coordinate so I assume the compilation
```

Comment:
This is enough for replacement, I agree. What about running times?

Reply:
Various compilers seems to be optimizing this differently. I'll do some tests and post them here.

Reply:
I think I'll remove this comment because this mostly reflects one compiler, it covers only a subset of the results and doesn't describe the overall picture. More results can be found in the comments of this PR.